types/yeoman-generator/index.d.ts

```diff
@@ -17,14 +17,30 @@ import { Observable } from 'rxjs';
 type Callback = (err: any) => void;
 
 declare namespace Generator {
-    interface Question extends inquirer.Question {
+    interface GeneratorQuestion {
         /**
          * whether to store the user's previous answer
          */
         store?: boolean;
     }
-    type Questions = Question | Question[] | Observable<Question>;
     type Answers = inquirer.Answers;
+    type Question<A extends Answers = Answers> = (
```

Comment:
I'd rather recommend you to use `GeneratorQuestion<A> extends inquirer.Question<A>` as this would cause the `Generator.Question`-type to look exactly the same.

Reply:
Yes, that would be ideal; however, it's not possible because of the union type.
```
An interface can only extend an object type or intersection of object types with statically
known members.ts(2312)
```
This is doable:
```ts
type Question<A extends Answers = Answers> = inquirer.Question<A>;
```
But this is not:
```ts
interface Question<A extends Answers = Answers> extends inquirer.Question<A> {}
```

Reply:
Actually, I have an idea...